src/main/kotlin/org/wfanet/panelmatch/client/privatemembership/JniQueryEvaluator.kt

```diff
@@ -14,29 +14,129 @@
 
 package org.wfanet.panelmatch.client.privatemembership
 
-import org.wfanet.panelmatch.common.loadLibraryFromResource
+import com.google.privatemembership.batch.Shared.EncryptedQueries
+import com.google.privatemembership.batch.Shared.EncryptedQueryResult
+import com.google.privatemembership.batch.Shared.Parameters
+import com.google.privatemembership.batch.Shared.PublicKey
+import com.google.privatemembership.batch.Shared.QueryMetadata
+import com.google.privatemembership.batch.encryptedQueryResult
+import com.google.privatemembership.batch.server.ApplyQueriesRequestKt.rawDatabase
+import com.google.privatemembership.batch.server.RawDatabaseShardKt.bucket
+import com.google.privatemembership.batch.server.Server.RawDatabaseShard
+import com.google.privatemembership.batch.server.SumCiphertextsRequestKt.summation
+import com.google.privatemembership.batch.server.applyQueriesRequest
+import com.google.privatemembership.batch.server.finalizeResultsRequest
+import com.google.privatemembership.batch.server.rawDatabaseShard
+import com.google.privatemembership.batch.server.sumCiphertextsRequest
+import com.google.protobuf.ByteString
+import rlwe.Serialization.SerializedSymmetricRlweCiphertext
 
 /** [QueryEvaluator] that calls into C++ via JNI. */
 class JniQueryEvaluator(private val parameters: QueryEvaluatorParameters) : QueryEvaluator {
+  private val privateMembershipParameters: Parameters by lazy {
+    Parameters.parseFrom(parameters.serializedPrivateMembershipParameters)
+  }
+
+  private val privateMembershipPublicKey: PublicKey by lazy {
+    PublicKey.parseFrom(parameters.serializedPublicKey)
+  }
 
   override fun executeQueries(
     shards: List<DatabaseShard>,
     queryBundles: List<QueryBundle>
   ): List<Result> {
-    TODO()
+    val request = applyQueriesRequest {
+      parameters = privateMembershipParameters
+      publicKey = privateMembershipPublicKey
+      finalizeResults = false
```

Comment:
Will it ever be the case that executeQueries receives the entire shard? If so, it will be more efficient to finalize and you may want to pass it as an argument to the function.